tsdb/blockwriter_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdb
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/util/testutil"
+)
+
+func TestBlockWriter(t *testing.T) {
+	ctx := context.Background()
+	outputDir, err := ioutil.TempDir(os.TempDir(), "output")
+	testutil.Ok(t, err)
+	w := NewBlockWriter(log.NewNopLogger(), outputDir, DefaultBlockDuration)
+	testutil.Ok(t, w.initHead())
+
+	// flush with no series results in error
+	_, err = w.Flush(ctx)
+	testutil.ErrorEqual(t, err, errors.New("no series appended, aborting"))
+
+	// add some series
+	app := w.Appender(ctx)
+	expectedLabelNames := "a"
+	l := labels.Labels{{Name: expectedLabelNames, Value: "b"}}
+	ts := int64(44)
+	_, err = app.Add(l, ts, 7)
+	testutil.Ok(t, err)
+	err = app.Commit()
+	testutil.Ok(t, err)
+	id, err := w.Flush(ctx)
+	testutil.Ok(t, err)
```

Comment:
Instead of adding only 1 sample, let's have some loop and ingest some N series with M samples in each series.

You could refer https://github.com/prometheus/prometheus/blob/2e512c823fed9e5ee4272053ba6e4fcbcf364d69/tsdb/head_test.go#L1938-L1964 which does something similar

Reply:
I can add that, but I'm curious what are we testing by adding more than one series?

Reply:
Nothing much TBH in this case. But in general >1 series activates some other operations, like the heap management and other index calculations during compaction.

Reply:
If it doesn't test much, then should we add it? The test here should be testing that the blockwriter functions as intended. Does the test without this modification do that?

Reply:
I added a 2nd series, sounds like that might be sufficient to trigger the extra stuff you were talking about